google/resource_compute_instance_template.go

```diff
@@ -420,6 +420,48 @@ func resourceComputeInstanceTemplate() *schema.Resource {
 	}
 }
 
+func resourceComputeInstanceTemplateCustomizeDiff(diff *schema.ResourceDiff, meta interface{}) error {
+	config := meta.(*Config)
+	project, err := getProjectFromDiff(diff, config)
+	if err != nil {
+		return err
+	}
+	numDisks := diff.Get("disk.#").(int)
+	for i := 0; i < numDisks; i++ {
+		key := fmt.Sprintf("disk.%d.source_image", i)
+		if diff.HasChange(key) {
+			old, new := diff.GetChange(key)
+			if old == "" || new == "" {
+				// no sense in resolving empty strings
+				err = diff.ForceNew(key)
+				if err != nil {
+					return err
+				}
+				continue
+			}
+			oldResolved, err := resolveImage(config, project, old.(string))
+			if err != nil {
+				return err
+			}
+			newResolved, err := resolveImage(config, project, new.(string))
+			if err != nil {
+				return err
+			}
+			if oldResolved != newResolved {
+				err = diff.ForceNew(key)
+				if err != nil {
+					return err
+				}
```

Comment:
shouldn't there be a continue here? otherwise the diff will get cleared on line 456

Reply:
Good catch.